design/oauth2-device-authorization-grant.md

```diff
@@ -0,0 +1,144 @@
+# OAuth 2.0 Device Authorization Grant
+
+* **Status**: Draft
+* **Author**: [wadahiro](https://github.com/wadahiro)
+* **JIRA**: [KEYCLOAK-7675](https://issues.jboss.org/browse/KEYCLOAK-7675)
+
+
+## Motivation
+
+[The OAuth 2.0 Device Authorization Grant](https://tools.ietf.org/html/draft-ietf-oauth-device-flow-15) is designed for internet-connected devices that have limited input capabilities or lack a suitable browser. The spec is still draft, but it has already been implemented by many major IdPs. By supporting this spec, we will be able to use Keycloak in more fields.
+
+## Endpoint
+
+### Device Authorization Endpoint
```

Comment:
I would suggest `/realms/{realm}/openid-connect/device/code`.

However, this proposal highlights an issue in our resource path format, which I hope we will improve with the new REST guidelines we are defining. Ideally, the path should `/realms/{realm}/oauth/device/code`.

Reply:
Why code and not auth? The step is called "Device Authorization Request", so auth makes more sense to me, as it is the equivalent of the regular OAuth auth endpoint.

Reply:
You are obtaining a code, both device and user codes.

Reply:
The regular authorization endpoint also returns a code which is called Authorization Code.  So I think it is better to match the existing endpoint.

Reply:
Let's stick with /auth/ to make it consistent with authorization code flow. Not only does it match the current endpoint, but it also makes more sense as it is the endpoint to initiate "auth", the fact that it returns a code is more an implementation detail.